schnorr_fun/src/frost_backup.rs

```diff
@@ -0,0 +1,198 @@
+//! Backup scheme for FROST secret shares
+//!
+//! # Description
+//!
+//! Based on https://bips.xyz/93
+//!
+//! ## Backup format (bech32 chars)
+//!
+//! human readable: "fr"            // (2)
+//! separator:      "1"             // (1)
+//! threshold:      u5,             // (1)
+//! identifier:     [u5; 4],        // (4)
+//! secret_share:   [u5; 52],       // (52)
+//! share_index:    u5 or [u5; 52], // (1 or 52)
+//! checksum:       [u5; 6],        // (6)
+//!
+//! ## Justification
+//!
+//! ### Threshold - (1 bech32 character)
+//! A threshold integer between 1 and 32.
+//! Note: a threshold of zero is considered invalid, and threshold is encoded as minus one.
+//!
+//! While this threshold will generally be ~small, in theory could be much higher than 32.
+//! Just doing up to 32 for now, but might be worth futureproofing also..
+//! We could also leave higher thresholds for a later backup version.
+//!
+//! ### Polynomial Identifier - (4)
+//! Hash of the polynomial coefficients and take the first 4 bech32 characters.
+//! This identifier will allow determination of secret share compatibility.
+//!
+//! It is possible to have two different polynomials that interpolate the same joint-secret,
+//! but shares on these different polynomials will not be compatible with one another.
+//!
+//! 32^4 = 2^20 ~> a collision rate of about one in a million.
```

Comment:
To be precise the probability of two different polynomials colliding is ~1 in a million.